crypto/bio/bss_conn.c

```diff
@@ -191,6 +191,9 @@ static int conn_state(BIO *b, BIO_CONNECT *c)
             break;
 
         case BIO_CONN_S_BLOCKED_CONNECT:
+            /* wait for socket being writable, before querying BIO_sock_error */
+            if (BIO_socket_wait(b->num, 0, time(NULL)) == 0)
+                break;
```

Comment:
Why do we need to do this?

Reply:
Apparently the connect is not yet completed when BIO_sock_error returns 0
here, and the ktls_enable call fails with ENOTCONN.  That is a race condition,
and does not happen every time.

The man page says

```
EINPROGRESS
              The socket is nonblocking and the connection cannot be completed
              immediately.   (UNIX domain sockets failed with EAGAIN instead.)
              It is possible to select(2) or poll(2) for completion by select‐
              ing the socket for writing.  After select(2) indicates writabil‐
              ity, use getsockopt(2) to read  the  SO_ERROR  option  at  level
              SOL_SOCKET to determine whether connect() completed successfully
              (SO_ERROR is zero) or unsuccessfully (SO_ERROR  is  one  of  the
              usual  error  codes  listed  here, explaining the reason for the
              failure).
```

But we are not doing the select here, probably that's the problem.
So this avoids the ktls_enable from failing, and makes the test case
use the client ktls successfully (modulo the possible #18276 issue).

Reply:
Ok